### DISCUSSION TITLE
Question: Is there any way to relax some constraints to force opf convergence?

### DISCUSSION THREAD

**Discussion by Crueda96** [e2nIEE/pandapower#1276] [Q&A]

**Comment by rbolgaryn** [comment]
Hi @Crueda96 ,

I know that matpower has a feature like this, that allows substitution of constraints with costs. But in pandapower we don't have it.
@MMajidi137 is it possible when using PowerModels?

Roman

**Comment by MMajidi137** [comment]
> Hi @Crueda96 ,
> 
> I know that matpower has a feature like this, that allows substitution of constraints with costs. But in pandapower we don't have it.
> @MMajidi137 is it possible when using PowerModels?
> 
> Roman

@Crueda96  Different possible relaxation are introduced in [PowerModels.jl](https://github.com/lanl-ansi/PowerModels.jl). However, the relaxation features are not available in the pandapower/develop current version. To add new features and tools related to [PowerModels.jl](https://github.com/lanl-ansi/PowerModels.jl), you need to checkout to the julia_pkg branch and follow the development structure in  [PandaModels.jl](https://github.com/e2nIEE/PandaModels.jl).

**Comment by MMajidi137** [comment]
@Crueda96 Did you solve the issue? you also can easily set the upper and lower bounds of the constraints that you want to relax in very big value( not exactly infinity).
